custom_components/aromalink_ha_integration/AromaLinkDeviceCoordinator.py

```
@@ -807,6 +838,55 @@ async def turn_on_off(self, state_to_set):
+                    )
+                    self.async_set_updated_data(optimistic_data)
+                    return True
+                elif response.status in [401, 403]:
+                    _LOGGER.warning(f"Authentication error on set_fan ({response.status}).")
+                    self.auth_coordinator.jsessionid = None
+                    return False
+                else:
+                    _LOGGER.error(
+                        f"Failed to control fan for device {self.device_id}: {response.status}")
+                    return False
+        except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)